src/lv_fonts/lv_font_template.c

```diff
@@ -0,0 +1,168 @@
+ *--------------------*/
+
+static uint16_t unicode_list_1 = {0x234, 0x295, 0x456};
+static uint16_t glyph_id_ofs_list_1 = {232, 545, 1466};
+
+static uint8_t glyph_id_ofs_list_3 = {32, 45, 66};
+
+
+/*Collect the unicode lists and glyph_id offsets*/
+static const lv_font_cmap_fmt_txt_t cmaps[] = {
+        { .range_start = 0x280, .range_length = 124,  .glyph_id_start = 223, .unicode_list = unicode_list_1,  .glyph_id_ofs_list = glyph_id_ofs_list_1},
+        { .range_start = 0x20,  .range_length = 112,  .glyph_id_start = 456, .unicode_list = NULL,            .glyph_id_ofs_list = NULL},
+        { .range_start = 0x560, .range_length = 7654, .glyph_id_start = 756, .unicode_list = NULL,            .glyph_id_ofs_list = glyph_id_ofs_list_3},
+
+};
+
+/*-----------------
+ *    KERNING
+ *----------------*/
+
+/*Left glyphs for kering*/
+static uint16_t kern_left_gylph_ids[] =
+{
+    0x21, 0x42, 0x53
+};
+
+/*Left glyphs for kering*/
+static uint16_t kern_right_gylph_ids[] =
+{
+    0x41, 0x342, 0x333
+};
+
+/* Kerning between the respective left and right glyphs
+ * 4.4 format which needs to scaled with `kern_scale`*/
+static int8_t kern_pair_values[] =
+{
+    23, -33, 55
+};
+
+/*Collect the kern pair's data in one place*/
+static lv_font_kern_pair_fmt_txt_t kern_pairs =
+{
+    .left_gylph_ids = kern_left_gylph_ids,
+    .right_gylph_ids = kern_right_gylph_ids,
+    .values = kern_pair_values,
+    .pair_cnt = 3,
+};
+
+/***** OR ******/
+
+/*Map glyph_ids to kern left classes*/
+static uint8_t kern_left_class_mapping[] =
+{
+        1, 2, 1, 1, 1, 1, 1, 1
+};
+
+/*Map glyph_ids to kern right classes*/
+static uint8_t kern_right_class_mapping[] =
+{
+        1, 2, 3, 2, 2, 3, 3, 1
+};
+
+/* Kern values between classes
+ * 4.4 format which needs to scaled with `kern_scale`*/
+static int8_t kern_class_values =
+{
+        44, -66, 24, -23, 56, 23
+};
+
+
+/*Collect the kern class' data in one place*/
+static lv_font_kern_classes_fmt_txt_t kern_classes =
+{
+        .class_pair_values = kern_class_values,
+        .left_class_mapping = kern_left_class_mapping,
+        .right_class_mapping = kern_right_class_mapping,
+        .left_class_cnt = 2,
+        .right_class_cnt = 3,
+};
+
+
+/*--------------------
+ *  ALL CUSTOM DATA
+ *--------------------*/
+
+/*Store all the custom data of the font*/
+static lv_font_dsc_fmt_txt_t font_dsc = {
+        .glyph_bitmap = gylph_bitmap,
+        .glyph_dsc = glyph_dsc,
+        .cmaps = cmaps,
+        .cmap_num = 3,
+        .bpp = 4,
+
+        .kerning_scale = 324,
+        .kern_dsc = kern_classes,
+        .kern_classes = 1,
+        /*** OR ***/
+        .kern_dsc = kern_pairs,
+        .kern_classes = 0,
+};
+
+
+/*-----------------
+ *  PUBLIC FONT
+ *----------------*/
+
+/*Initialize a public general font descriptor*/
+lv_font_t font_name = {
+    .unicode_first = 0x20,     /*First Unicode letter in this font*/
+    .unicode_last = 0x3F45,    /*Last Unicode letter in this font*/
```

Comment:
2 props above seems not needed - tail from previous format with chained blocks.

Reply:
Removed